services/otp/otp.go

```diff
@@ -0,0 +1,109 @@
+package otp
+
+import (
+	"encoding/json"
+	"net/http"
+
+	sa "github.com/secureauthcorp/saidp-sdk-go"
+)
+
+/*
+**********************************************************************
+*   @author scox@secureauth.com
+*
+*  Copyright (c) 2017, SecureAuth
+*  All rights reserved.
+*
+*    Redistribution and use in source and binary forms, with or without modification,
+*    are permitted provided that the following conditions are met:
+*
+*    1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
+*
+*    2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer
+*    in the documentation and/or other materials provided with the distribution.
+*
+*    3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote products derived
+*    from this software without specific prior written permission.
+*
+*    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+*    THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+*    CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+*    PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+*    LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+*    EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+**********************************************************************
+ */
+
+const endpoint = "/api/v1/otp/validate"
+
+// Response :
+//	Response struct that will be populated after the post request.
+type Response struct {
+	Status       string         `json:"status,omitempty"`
+	Message      string         `json:"message,omitempty"`
+	UserID       string         `json:"user_id,omitempty"`
+	HTTPResponse *http.Response `json:"-,omitempty"`
+}
+
+// Request :
+//	Request struct to build the required post parameters.
+// Fields:
+//	[Required] UserId: The username that you want to validate an otp for.
+//	[Required] Domain: The domain of the user you want to validate an otp for.
+//	[Required] OTP: The OTP you want to validate.
+type Request struct {
```

Comment:
Are any of these omit-able? Meaning do I have to have a Domain in the strut? I know go will place the default value for the string type (""), but should it be omit-able on the marshaling of the JSON object? Not that it matters to much.